### ISSUE TITLE
Joining spectators from team red makes a blue death effect

### ISSUE THREAD

**Issue by fokkonaut** [teeworlds/teeworlds#2400]

**Comment by fokkonaut** [comment]
Not a client bug

**Comment by fokkonaut** [comment]
sure its a client bug?

**Comment by oy** [comment]
Yeah, kill messages just send client ID's, no team stuff.

**Comment by fokkonaut** [comment]
Its not a kill message, its the CreateDeath() function

**Comment by oy** [comment]
Ah, you mean when the character explodes?

**Comment by fokkonaut** [comment]
Yes

**Comment by Dune-jr** [comment]
Yeah, I can reproduce that. I think it's a server issue, changing team before killing the character

**Comment by fokkonaut** [comment]
Thats my guess too

**Comment by oy** [comment]
The events are just sent with snapshots, team change is sent instant. So yeah timing thing.

**Comment by AssassinTee** [comment]
Description says it's blue, but it's purple, because the spectator teamcolor is purple.
If it was blue, there would be an extra bug.

**Comment by fokkonaut** [comment]
Spectators dont get their skins changed. Its obviously blue, because the player somehow gets put into team blue before, then gets killed for the teamchange so the client thinks the tee is in team blue and makes the blue death effect from your tee because it got blue teamcolor applied.

**Comment by fokkonaut** [comment]
> I propose adding the team to the message or a color to the message.
this is just a small bug, no need for a "huge" change

**Comment by AssassinTee** [comment]
No it's purple**.**

Assa joined spectators:
![](https://i.imgur.com/muShMZO.png)

Assa joined blue:
![](https://i.imgur.com/SCyxU3v.png)

**Comment by fokkonaut** [comment]
I am colorblind, sorry, for me it looked like blue. But why is it purple?

**Comment by AssassinTee** [comment]
> Spectators have their own teamcolor! Ever spawned a spectator?

.

**Comment by fokkonaut** [comment]
How would I spawn a spectator? For me they always looked normal, probably because I have a non-team gamemode?

**Comment by AssassinTee** [comment]
> How would I spawn a spectator? For me they always looked normal, probably because I have a non-team gamemode?

remove this line: https://github.com/teeworlds/teeworlds/blob/de5859b3717406f9f0e728506ad0ef15d4e87337/src/game/server/player.cpp#L343

**Comment by fokkonaut** [comment]
Thats why I never saw this before, my mod is a non team gamemode, but i also send team changes to spectator (for /pause for example)
